apps/showcase/src/services/webcontainer/webcontainer.service.ts

```diff
@@ -0,0 +1,133 @@
+        .map(([p, n]) => convertTreeRec(p, n))
+      : undefined
+  };
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class WebcontainerService {
+  public readonly runner = new WebContainerRunner();
+  private readonly monacoTree = new BehaviorSubject<MonacoTreeElement[]>([]);
+
+  public monacoTree$ = this.monacoTree.pipe(
+    distinctUntilChanged((a, b) => JSON.stringify(a) === JSON.stringify(b))
+  );
+  public isReady$ = this.monacoTree$.pipe(
+    map((tree) => tree.length > 0)
+  );
+
+  /**
+   * Get the Monaco file tree from the given root path
+   * @param rootPath
+   * @private
+   */
+  private async getMonacoTree(rootPath: string): Promise<MonacoTreeElement[]> {
+    const instance = await this.runner.instance;
+    // eslint-disable-next-line no-console
+    console.log('Monaco - start', rootPath);
+    const tree = await this.getFilesTreeFromContainer(instance, rootPath);
+    // eslint-disable-next-line no-console
+    console.log('Monaco - ', rootPath, tree);
+    return Object.entries(tree)
+      .reduce(
+        (acc: MonacoTreeElement[], [path, node]) =>
+          acc.concat(convertTreeRec(path, node)),
+        []
+      );
+  }
+
+  /**
+   * Get the file tree from the path of the File System of the given WebContainer instance
+   * @param instance
+   * @param path
+   * @private
+   */
+  private async getFilesTreeFromContainer(instance: WebContainer, path = '/') {
+    return await getFilesTree([{
+      path,
+      isDir: true
+    }], instance.fs as FileSystem, EXCLUDED_FILES_OR_DIRECTORY);
+  }
+
+  /**
+   * Load a new project: mount the files in the dedicated folder, update the monaco tree and watch the folder updates and
+   * run the initialization commands
+   * @param files
+   * @param commands
+   * @param exerciseName
+   */
+  public async loadProject(files: FileSystemTree, commands: string[], exerciseName: string) {
+    this.monacoTree.next([]);
+    this.runner.registerTreeUpdateCallback(async () => {
+      const tree = await this.getMonacoTree(`/${exerciseName}`);
+      this.monacoTree.next(tree);
+    });
+    return this.runner.runProject(files, commands, exerciseName);
+  }
+
+  /**
+   * Writes a file with the provided content to the given path
+   * @param file
+   * @param content
+   */
+  public async writeFile(file: string, content: string) {
+    const instance = await this.runner.instance;
+
+    return instance.fs.writeFile(file, content);
+  }
+
+  /**
+   * Reads the file at the given path
+   * @param file
+   */
+  public async readFile(file: string): Promise<string> {
+    const instance = await this.runner.instance;
+
+    return instance.fs.readFile(file, 'utf-8');
+  }
+
+  /**
+   * Determine if the file path entry is a file
+   * @param filePath
+   */
+  public async isFile(filePath: string) {
+    const instance = await this.runner.instance;
+    const parent = `${!filePath.startsWith('/') ? '/' : ''}${filePath}`
```

Comment:
from what I understand, this method only supports absolute paths, it could be worth mentioning in the doc that relative paths are not supported